TeamCode/src/main/java/org/firstinspires/ftc/teamcode/rework/vision/RingStackLocator.java

```diff
@@ -0,0 +1,192 @@
+package org.firstinspires.ftc.teamcode.rework.vision;
+
+import org.opencv.core.*;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.imgproc.Moments;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.opencv.imgproc.Imgproc.*;
+
+public class RingStackLocator {
+
+    static final Scalar RED = new Scalar(0, 0, 255);
+    static final Scalar GREEN = new Scalar(0, 255, 0);
+    static final Scalar BLUE = new Scalar(255, 0, 0);
+
+    public enum TARGET_ZONE {
+            TARGET_ZONE_A,
+        TARGET_ZONE_B,
+        TARGET_ZONE_C
+    }
+
+
+    private static double[] roots(double a, double b, double c) {
+        double[] roots = new double[2]; //This is now a double, too.
+        double discriminator = Math.sqrt(Math.pow(b, 2) - 4 * a * c);
+        roots[0] = (-b + discriminator) / (2 * a);
+        roots[1] = (-b - discriminator) / (2 * a);
+        return roots;
+    }
+    public static double find_ring_angle(double width, double height) {
```

Comment:
uh could you please leave comments so I don't have to find the intuition behind all this math, much easier for me to read.

Reply:
will remove find_ring_angle is not useful for now